### ISSUE TITLE
SvgImage.toImageScaled produces only uneven wide images (for uneven input)

### ISSUE THREAD

**Issue by JFBM** [hervegirod/fxsvgimage#13]
I'm trying to convert an SVG of a QR-Code to an Image. The QR-Code has a size of 25 (which is loaded correctly).
Ultimately there should be several sizes, all powers of 2, to be usable as icons, but the problem happens with any size.

If I enter a scale-factor that would produce the correct `width`, it instead returns an image with `width+1`.
E.g. `qr.toImageScaled(64 / qr.getWidth)` will return an Image with `width` and `height` of `65` (instead of the desired `64`).
Furthermore, outsmarting the algorithm doesn't work. `qr.toImageScaled(63 / 25)` produces an image of width `63`.

Preliminary testing shows that it snaps to the next uneven number of pixels, rounded up.

Is that a limitation of the rendering? I've seen you use a Scene to render a snapshot into an image, but am currently too tired to debug that far into it.

**Comment by hervegirod** [comment]
Hello, do you have an example? I should be able to create one myself I think, but if you are able to send me one example, I would be sure that I fix your problem completely :)

**Comment by hervegirod** [comment]
Hello, I am able to reproduce the problem. Strangely, this works correctly (supposing an input svg of size 50):
```
      SVGImage result = SVGLoader.load(url, 25);
      Image img = result.toImage();
```
while this is off by one pixel (result has a size of 26 rather than 25):
```
      SVGImage result = SVGLoader.load(url);
      Image img = result.toImage(25);
```
It might be a JavaFX problem. I think that if I should internally resize the JavaFX content itself before creating the snapshot and not scale using the JavaFX snapshot parameters.

**Comment by hervegirod** [comment]
I think that I fixed it on master.

**Comment by hervegirod** [comment]
It might be fixed in the 1.0b2 release

**Comment by hervegirod** [comment]
Tagged as fixed
